### ISSUE TITLE
Consider deferred bundleization at first storage service

### ISSUE THREAD

**Issue by jphickey** [nasa/bplib#108]
Currently the input data is turned into a bundle at the time the application calls "bplib_send", and the bundle is temporarily stored in memory.  This bundle is then routed to a storage service for retention, which offloads it to some form of persistent storage (at least after #90 is implemented).

This basically entails making two copies, first into temporary storage then into the persistent storage.  This could potentially be reduced, if the bundleization step was deferred until it arrived at storage (or at least the payload was sent as a pointer rather than a copy), then it could be only a single copy directly into persistent storage.
